File: src/main/java/com/mbayou/kick4k/moderation/DeleteModerationBansRequest.java
package com.mbayou.kick4k.moderation;

import com.fasterxml.jackson.annotation.JsonProperty;

public class DeleteModerationBansRequest {
    private final Integer broadcasterUserId;
    private final Integer userId;

    public DeleteModerationBansRequest(@JsonProperty("broadcaster_user_id") Integer broadcasterUserId,
                                       @JsonProperty("user_id") Integer userId) {
        this.broadcasterUserId = broadcasterUserId;
        this.userId = userId;
    }

    public Integer getBroadcasterUserId() {
        return broadcasterUserId;
    }

    public Integer getUserId() {
        return userId;
    }

    public static Builder builder() {
        return new Builder();
    }

    public static class Builder {
        private Integer broadcasterUserId;
        private Integer userId;

        public Builder broadcasterUserId(Integer broadcasterUserId) {
            this.broadcasterUserId = broadcasterUserId;
            return this;
        }

        public Builder userId(Integer userId) {
            this.userId = userId;
            return this;
        }

        public DeleteModerationBansRequest build() {
            if (broadcasterUserId == null) {
                throw new IllegalStateException("BroadcasterUserId is required");
            }

            if (userId == null) {
                throw new IllegalStateException("UserId is required");
            }

            return new DeleteModerationBansRequest(broadcasterUserId, userId);
        }
    }
}

File: src/main/kotlin/com/mbayou/kick4k/moderation/DeleteModerationBansRequest.kt
package com.mbayou.kick4k.moderation

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty

data class DeleteModerationBansRequest @JsonCreator constructor(
    @JsonProperty("broadcaster_user_id") val broadcasterUserId: Int,
    @JsonProperty("user_id") val userId: Int,
) {
    companion object {
        @JvmStatic
        fun builder(): Builder = Builder()
    }

    class Builder {
        private var broadcasterUserId: Int? = null
        private var userId: Int? = null

        fun broadcasterUserId(broadcasterUserId: Int) = apply { this.broadcasterUserId = broadcasterUserId }
        fun userId(userId: Int) = apply { this.userId = userId }

        fun build(): DeleteModerationBansRequest {
            val broadcasterId = requireNotNull(broadcasterUserId) { "BroadcasterUserId is required" }
            val userIdValue = requireNotNull(userId) { "UserId is required" }
            return DeleteModerationBansRequest(broadcasterId, userIdValue)
        }
    }
}
